### ISSUE TITLE
Less common chunks if rolldown doesn't need to preserve the exports of entires

### ISSUE THREAD

**Issue by underfin** [rolldown/rolldown#1842]
### Reproduction link or steps

The demo 

main.js is entry
``` .js
import { a } from './file' 
import('./file')
```

file.js
``` .js
export const a = 1;
```

### What is expected?

Output should be same with [esbuild](https://esbuild.github.io/try/#YgAwLjIzLjAALS1idW5kbGUgLS1mb3JtYXQ9Y2pzAGUAZW50cnkuanMAaW1wb3J0IHthIH0gZnJvbSAnLi9maWxlJwppbXBvcnQoJy4vZmlsZScpAABmaWxlLmpzAGV4cG9ydCBjb25zdCBhID0gMQ)

### What is actually happening?

The rolldown output

file.mjs

```js
import { a } from "./file~1.mjs";

export { a };
```
 file~1.mjs

```js

//#region file.js
const a = 1;

//#endregion
export { a };
```
main.mjs

```js
import "./file~1.mjs";

//#region main.js
import("./file.mjs");

//#endregion
```

### System Info

```Shell
-
```

**Comment by hyf0** [comment]
You didn't enable `--splitting` in esbuild.

It seems looks the same as esbuild's.

https://esbuild.github.io/try/#YgAwLjIzLjAALS1idW5kbGUgLS1mb3JtYXQ9Y2pzIC0tc3BsaXR0aW5nIC0tb3V0ZGlyPWRpc3QgLS1mb3JtYXQ9ZXNtAGUAZW50cnkuanMAaW1wb3J0IHthIH0gZnJvbSAnLi9maWxlJwppbXBvcnQoJy4vZmlsZScpAABmaWxlLmpzAGV4cG9ydCBjb25zdCBhID0gMQ ?

**Comment by underfin** [comment]
The rollup behavior https://rollupjs.org/repl/?version=4.19.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmElMjAlN0QlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhhKSU1Q25pbXBvcnQoJy4lMkZmaWxlJyklNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGElMjAlM0QlMjAxJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTJDJTIybmFtZSUyMiUzQSUyMmZpbGUuanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==

**Comment by underfin** [comment]
> You didn't enable `--splitting` in esbuild.
> 
> It seems looks the same as esbuild's.
> 
> https://esbuild.github.io/try/#YgAwLjIzLjAALS1idW5kbGUgLS1mb3JtYXQ9Y2pzIC0tc3BsaXR0aW5nIC0tb3V0ZGlyPWRpc3QgLS1mb3JtYXQ9ZXNtAGUAZW50cnkuanMAaW1wb3J0IHthIH0gZnJvbSAnLi9maWxlJwppbXBvcnQoJy4vZmlsZScpAABmaWxlLmpzAGV4cG9ydCBjb25zdCBhID0gMQ ?

Yeah. I missed it. Here is different with rollup and esbuild, the vite will give a warning for the rollup behavior, see [here](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/reporter.ts#L120-L143). cc @yyx990803

**Comment by hyf0** [comment]
Yeah. Rollup might emits better output here. But this still doesn't seem to be a __bug__. The current behavior is same with esbuild.

**Comment by hyf0** [comment]
With current information I get, I don't see any urgent request to support such optimization. So I intend not to introduce this feature/complexity now before finishing the css support.

**Comment by hyf0** [comment]
Seems a very neat/delicate optimization: https://rollupjs.org/repl/?version=4.19.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmElMjAlN0QlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhhKSU1Q25pbXBvcnQoJy4lMkZmaWxlJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbmltcG9ydCgnLiUyRmZpbGUyJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBhJTIwJTNEJTIwMSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHN0YXIlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhzdGFyKSUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlMi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJpbmxpbmVEeW5hbWljSW1wb3J0cyUyMiUzQWZhbHNlJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==

**Comment by hyf0** [comment]
The key might be if the entry module has exports/if we need preserve the exports of entries

https://rollupjs.org/repl/?version=4.19.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmElMjAlN0QlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhhKSU1Q25pbXBvcnQoJy4lMkZmaWxlJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbmltcG9ydCgnLiUyRmZpbGUyJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbiU1Q25leHBvcnQlMjAlN0IlMjBhJTIwJTdEJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBhJTIwJTNEJTIwMSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHN0YXIlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhzdGFyKSUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlMi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJpbmxpbmVEeW5hbWljSW1wb3J0cyUyMiUzQWZhbHNlJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==

**Comment by underfin** [comment]
The entry point as othe entry import also has the problem.

The [esbuild](https://esbuild.github.io/try/#YgAwLjIzLjAALS1idW5kbGUgLS1vdXRkaXI9YSAtLXNwbGl0dGluZyAtLWZvcm1hdD1lc20AZQBlbnRyeS5qcwBpbXBvcnQgJy4vZmlsZScAZQBmaWxlLmpzAGNvbnNvbGUubG9nKCcnKQ), the [rollup](https://rollupjs.org/repl/?version=4.19.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmElMjAlN0QlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhhKSU1Q25pbXBvcnQoJy4lMkZmaWxlJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbmltcG9ydCgnLiUyRmZpbGUyJykudGhlbihleHBvcnRzJTIwJTNEJTNFJTIwY29uc29sZS5sb2coZXhwb3J0cy5hJTIwJTNEJTNEJTIwYSkpJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBhJTIwJTNEJTIwMSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHN0YXIlMjBmcm9tJTIwJy4lMkZmaWxlJyU1Q25jb25zb2xlLmxvZyhzdGFyKSUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJmaWxlMi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJpbmxpbmVEeW5hbWljSW1wb3J0cyUyMiUzQWZhbHNlJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==)

**Comment by sapphi-red** [comment]
This issue should be covered by

- https://github.com/rolldown/rolldown/issues/4904
- https://github.com/rolldown/rolldown/issues/4905
